addons/payment_buckaroo/models/payment_acquirer.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-'8' "-*-"
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+import logging
+import urllib
+import urlparse
+
+from odoo.addons.payment_buckaroo.controllers.main import BuckarooController
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    brq_websitekey = fields.Char(string='WebsiteKey', required_if_provider='buckaroo')
+    brq_secretkey = fields.Char(string='SecretKey', required_if_provider='buckaroo')
+
+    def _get_buckaroo_urls(self, environment):
+        """ Buckaroo URLs
+        """
+        if environment == 'prod':
+            return {
+                'buckaroo_form_url': 'https://checkout.buckaroo.nl/html/',
+            }
+        else:
+            return {
+                'buckaroo_form_url': 'https://testcheckout.buckaroo.nl/html/',
+            }
+
+    @api.model
+    def _get_providers(self):
+        providers = super(PaymentAcquirer, self)._get_providers()
+        providers.append(['buckaroo', 'Buckaroo'])
+        return providers
+
+    def _buckaroo_generate_digital_sign(self, inout, values):
+        """ Generate the shasign for incoming or outgoing communications.
+        :param string inout: 'in' (odoo contacting buckaroo) or 'out' (buckaroo
+                             contacting odoo).
+        :param dict values: transaction values
+
+        :return string: shasign
+        """
+        assert inout in ('in', 'out')
+        assert self.provider == 'buckaroo'
+
+        keys = "add_returndata Brq_amount Brq_culture Brq_currency Brq_invoicenumber Brq_return Brq_returncancel Brq_returnerror Brq_returnreject brq_test Brq_websitekey".split()
+
+        def get_value(key):
+            if values.get(key):
+                return values[key]
+            return ''
+
+        if inout == 'out':
+            for key in values.keys():
```

Comment:
Python dicts are automatically iterated by key, so the `.keys()` call is not useful.